cmd/snap-bootstrap/triggerwatch/triggerwatch_test.go

```diff
@@ -233,3 +233,110 @@ func (s *triggerwatchSuite) TestUdevEventNoKeyEvent(c *C) {
 		c.Assert(md.closeCalls, Equals, 1)
 	})
 }
+
+func (s *triggerwatchSuite) TestWaitMoreKeyboards(c *C) {
+	nodepath := "/dev/input/event0"
+	devpath := "/devices/SOMEBUS/input/input0/event0"
+
+	md := &mockTriggerDevice{}
+	md2 := &mockTriggerDevice{ev: &triggerwatch.KeyEvent{}}
+	mi := &mockTrigger{
+		d: md,
+		unlistedDevices: map[string]*mockTriggerDevice{
+			nodepath: md2,
+		},
+	}
+	restore := triggerwatch.MockInput(mi)
+	defer restore()
+
+	uevents := make(chan netlink.UEvent)
+	restore = triggerwatch.MockUEventChannel(uevents)
+	defer restore()
+
+	type activeTimer struct {
+		duration time.Duration
+		time     chan time.Time
+	}
+
+	start := time.Now()
+	currentNow := start
+	currentMutex := &sync.Mutex{}
+
+	var timers []*activeTimer
+	restore = triggerwatch.MockTimeAfter(func(d time.Duration) <-chan time.Time {
+		if d < 0 {
+			c.Errorf("Timer with negative duration")
+			ret := make(chan time.Time, 1)
+			currentMutex.Lock()
+			defer currentMutex.Unlock()
+			ret <- currentNow
+			return ret
+		} else {
+			currentMutex.Lock()
+			defer currentMutex.Unlock()
+			durationFromStart := currentNow.Add(d).Sub(start)
+			timer := &activeTimer{
+				duration: durationFromStart,
+				time:     make(chan time.Time, 1),
+			}
+			timers = append(timers, timer)
+			return timer.time
+		}
+	})
+	defer restore()
+
+	advanceTo := func(d time.Duration) {
+		currentMutex.Lock()
+		defer currentMutex.Unlock()
+		currentNow = start.Add(d)
+		var remaining []*activeTimer
+		for _, timer := range timers {
+			if timer.duration > d {
+				remaining = append(remaining, timer)
+			} else {
+				timer.time <- currentNow
+			}
+		}
+		timers = remaining
+	}
+
+	waitResult := make(chan error)
+	go func() {
+		err := triggerwatch.Wait(10*time.Second, 2*time.Second)
+		waitResult <- err
+	}()
+
+	// Flush go routines to block
+	<-time.After(time.Second)
+
+	advanceTo(3 * time.Second)
```

Comment:
is the idea that this will trigger the device timeout that is passed as 2s above? maybe a comment would be good if that is the case

Reply:
Yes. Something more than 2 less than 10. I can add a comment.

Reply:
thanks